model/aro_hcp/v1alpha1/managed_identities_requirements_type.model

```diff
@@ -0,0 +1,108 @@
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Representation of managed identities requirements.
+// When creating ARO-HCP Clusters, the end-users will need to pre-create the set of Managed Identities
+// required by the clusters.
+// The set of Managed Identities that the end-users need to precreate is not static and depends on 
+// several factors:
+// (1) The OpenShift version of the cluster being created. 
+// (2) The functionalities that are being enabled for the cluster. Some Managed Identities are not
+// always required but become required if a given functionality is enabled.
+// Additionally, the Managed Identities that the end-users will need to precreate will have to have a 
+// set of required permissions assigned to them which also have to be returned to the end users.
+class ManagedIdentitiesRequirements {
+  // The control plane operators managed identities requirements
+  ControlPlaneOperatorsIdentities []ControlPlaneOperatorIdentity
+  // The data plane operators managed identities requires
+  DataPlaneOperatorsIdentities []DataPlaneOperatorIdentity
+}
+
+struct ControlPlaneOperatorIdentity {
+  // Indicates the whether the identity is always required or not.
+  // "always" means that the identity is always required
+  // "on_enablement" means that the identity is only required when a functionality 
+  // that leverages the operator is enabled. 
+  // Possible values are ("always", "on_enablement")
+  Required String
+
+   // the name of the control plane operator that needs the identity
+   Name String
+
+   // The field is a string and it is of format X.Y.
+   // Not specifying it indicates support for this operator in all Openshift versions,
+   // starting from min_openshift_version if min_openshift_version is defined. 
+   @json(name="max_openshift_version")
+   MaxOpenShiftVersion String
+
+   // The field is a string and it is of format X.Y.
+   // Not specifying it indicates support for this operator in all Openshift versions,
+   // or up to max_openshift_version, if defined. 
+   @json(name="min_openshift_version")
+   MinOpenShiftVersion String   
+
+   // A list of roles that are required by the operator 
+   RoleDefinitions []RoleDefinition
+}
+
+struct DataPlaneOperatorIdentity {
+   // Indicates the whether the identity is always required or not
+   // "always" means that the identity is always required
+   // "on_enablement" means that the identity is only required when a functionality 
+   // that leverages the operator is enabled.
+   // Possible values are ("always", "on_enablement")
+   Required String
+
+   // the name of the data plane operator that needs the identity
+   Name String
```

Comment:
Now that the containing section has been renamed to DataPlaneOperatorsIdentities we should rename this to OperatorName as Name could be interpreted as "the identity name".

The same applies to the control plane ones

Reply:
make sense.

I've changed it.